src/evm/protocol/uniswap_v4/utils.rs

```diff
@@ -0,0 +1,27 @@
+use alloy_primitives::Address;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[repr(u8)]
+pub enum HookOptions {
+    AfterRemoveLiquidityReturnsDelta = 0,
+    AfterAddLiquidityReturnsDelta = 1,
+    AfterSwapReturnsDelta = 2,
+    BeforeSwapReturnsDelta = 3,
+    AfterDonate = 4,
+    BeforeDonate = 5,
+    AfterSwap = 6,
+    BeforeSwap = 7,
+    AfterRemoveLiquidity = 8,
+    BeforeRemoveLiquidity = 9,
+    AfterAddLiquidity = 10,
+    BeforeAddLiquidity = 11,
+    AfterInitialize = 12,
+    BeforeInitialize = 13,
+}
```

Comment:
Are these the only options? How are these extracted/categorized? What if a hook wants to do multiple of these, or something not included?

Reply:
yes these are the only options. I'm not sure how but this is all added when making the hook address. but we are only interested in decoding this (I took this from [here](https://github.com/shuhuiluo/uniswap-v4-sdk-rs/blob/main/src/utils/hook.rs#L69))

Reply:
ohhh so basically we reject all hooks that dont fall into these